ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/util/PDETextHelper.java

```diff
@@ -283,13 +281,49 @@ private static boolean isValidTagAttributeList(String text) {
 		// " att1="value1" att2="value2"
 		// " att1="value1" att2="value2 /"
 		// " att1="value1"
+		while (!text.isBlank()) {
+			int idx = text.indexOf('=');
+			if (idx < 0) {
+				return text.trim().equals("/"); //$NON-NLS-1$
+			}
+			String key = text.substring(0, idx).trim();
+			if (!isValidAttributeKey(key)) {
+				return false;
+			}
+			String remaining = text.substring(idx + 1).trim();
+			if (!remaining.startsWith("\"")) { //$NON-NLS-1$
+				return false;
+			}
+			int end = remaining.indexOf('"', 1);
+			if (end < 0) {
+				return false;
+			}
+			text = remaining.substring(end + 1);
+		}
+		return true;
+	}
+
+	private static boolean isValidAttributeKey(String key) {
+		if (key.isEmpty()) {
+			return false;
+		}
+		int length = key.length();
+		for (int i = 0; i < length; i++) {
+			char c = key.charAt(i);
+			if (!isValidAttributeKeyChar(c)) {
+				return false;
+			}
+		}
+		return true;
```

Comment:
Would a stream-pipeline fit here?
```suggestion
		return key.chars().allMatch(PDETextHelper::isValidAttributeKeyChar);
```
With the suggestions below one could maybe even inline the isValidAttributeKeyChar method?

Reply:
What about this part?